File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        int n=nums.size();
        for(int i=0;i<n;i++){
            int more=target-nums[i];
            if(mp.find(more)!=mp.end())
            return {mp[more],i};
            else
            mp[nums[i]]=i;
        }
        return {};
    }
};

File: 0001-two-sum/solution.go
func twoSum(nums []int, target int) []int {
    mp:= make(map[int]int)
    var res []int
    for i, num := range nums {
        if _,ok:=mp[target-num];ok {
            res=append(res,i)
            res=append(res,mp[target-num])
            return res
        } else {
            mp[num]=i
        }
    }
    return res
}
